source/common/io/io_uring_worker_impl.cc

```diff
@@ -611,5 +638,41 @@ void IoUringServerSocket::submitWriteOrShutdownRequest() {
   }
 }
 
+IoUringClientSocket::IoUringClientSocket(os_fd_t fd, IoUringWorkerImpl& parent,
+                                         Event::FileReadyCb cb, uint32_t write_timeout_ms,
+                                         bool enable_close_event)
+    : IoUringServerSocket(fd, parent, cb, write_timeout_ms, enable_close_event) {}
+
+void IoUringClientSocket::connect(const Network::Address::InstanceConstSharedPtr& address) {
+  // Reuse read request since there is no read on connecting and connect is cancellable.
```

Comment:
I wasn't sure reuse the `read_req_` is good choice. That may tricky us in the future.

Reply:
read and connect are consistent in the current logical handling (in close, cancel). Given the simplicity of the code, we are going to maintain the status after discussing it offline.

Reply:
yes, let us keep it.